Exiled.CustomRoles/API/Extensions.cs

```diff
@@ -0,0 +1,41 @@
+// -----------------------------------------------------------------------
+// <copyright file="Extensions.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomRoles.API
+{
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+
+    using Exiled.API.Features;
+    using Exiled.CustomRoles.API.Features;
+
+    using UnityEngine;
+
+    /// <summary>
+    /// A collection of API methods.
+    /// </summary>
+    public static class Extensions
+    {
+        /// <summary>
+        /// Gets a <see cref="ReadOnlyCollection{T}"/> of the player's current custom roles.
+        /// </summary>
+        /// <param name="player">The <see cref="Player"/> to check for roles.</param>
+        /// <returns>A <see cref="ReadOnlyCollection{T}"/> of all current custom roles.</returns>
+        public static ReadOnlyCollection<CustomRole> GetCustomRoles(this Player player)
+        {
+            List<CustomRole> roles = new List<CustomRole>();
+            Component[] components = player.GameObject.GetComponents(typeof(CustomRole));
+            foreach (Component component in components)
+            {
+                if (component is CustomRole role)
+                    roles.Add(role);
+            }
```

Comment:
Why is this check needed? Doesn't unity ensure they all are custom roles?

Reply:
probably but GetComponents(typeof()) returns a Component[] not a CustomRole[] so I need to cast them before adding them to the list anyways.

Also these probably won't remain components since that makes setting up configs for them more annoying than if the CustomRole itself was able to be serialized as a config object like CustomItems are